src/cloudsc_fortran/dwarf_cloudsc.F90

```diff
@@ -87,13 +103,143 @@ PROGRAM DWARF_CLOUDSC
      & GLOBAL_STATE%PFSQLF,   GLOBAL_STATE%PFSQIF ,  GLOBAL_STATE%PFCQNNG,  GLOBAL_STATE%PFCQLNG, &
      & GLOBAL_STATE%PFSQRF,   GLOBAL_STATE%PFSQSF ,  GLOBAL_STATE%PFCQRNG,  GLOBAL_STATE%PFCQSNG, &
      & GLOBAL_STATE%PFSQLTUR, GLOBAL_STATE%PFSQITUR, &
-     & GLOBAL_STATE%PFPLSL,   GLOBAL_STATE%PFPLSN,   GLOBAL_STATE%PFHPSL,   GLOBAL_STATE%PFHPSN &
-     & )
+     & GLOBAL_STATE%PFPLSL,   GLOBAL_STATE%PFPLSN,   GLOBAL_STATE%PFHPSL,   GLOBAL_STATE%PFHPSN, &
+     & YDOMCST, YDOETHF, YDECLDP, YDEPHLI)
 
 ! Validate the output against serialized reference data
 CALL GLOBAL_STATE%VALIDATE(NPROMA, NGPTOT, NGPTOTG)
 
 ! Tear down MPI environment
 CALL CLOUDSC_MPI_END()
 
+CONTAINS
+
+SUBROUTINE INITIALIZE_PARAMETER_DATATYPES( YDOMCST, YDOETHF, YDECLDP, YDEPHLI )
+
+USE YOMCST   , ONLY : RETV, RG  ,RCPD                        ,&
+ & RLVTT    ,RLSTT    ,RLMLT    ,RTT     ,RD                 ,&
+   RA, RPI, RV
+USE YOETHF   , ONLY : R2ES     ,R3LES    ,R3IES    ,R4LES    ,&
+ & R4IES    ,R5LES    ,R5IES    ,R5ALVCP  ,R5ALSCP           ,&
+ & RALVDCP  ,RALSDCP  ,RTWAT    ,RTICE    ,RTICECU           ,&
+ & RTWAT_RTICE_R      ,RTWAT_RTICECU_R                       ,&                          
+ & RALFDCP, RKOOP1, RKOOP2
+USE YOECLDP    , ONLY : YRECLDP
+USE YOEPHLI    , ONLY : YREPHLI
+
+TYPE(TOMCST)      ,INTENT(INOUT) :: YDOMCST
+TYPE(TOETHF)      ,INTENT(INOUT) :: YDOETHF
+TYPE(TECLDP)      ,INTENT(INOUT) :: YDECLDP
+TYPE(TEPHLI)      ,INTENT(INOUT) :: YDEPHLI
+
+YDECLDP%RKCONV          = YRECLDP%RKCONV
```

Comment:
Should these loader routines not live in the respective modules, like YOETHF, YOEMCST, etc.? We already have global loader functions in there, and we could probably add type-specific ones there (either `YDETHF%INITIALIZE()` or add something like `YOETHF_LOAD_PARAMETERS(YDETHF)` to the existing loader routine? This would avoid polluting the high-level entry routines with this.